### PR TITLE
dconf2nix: 0.0.7 -> 0.0.8

### PR DESCRIPTION
**Pull request by gvolpe**
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix:8]: Please use fetchFromGitHub.

File: pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix:10]:
```suggestion
    rev = "v${version}";
    sha256 = "19jk3p0ys3lbqk21lm37a7alkg2vhnmkvcffjqfxrw8p4737hxid";
```

File: pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix:4]:
```suggestion
}:

mkDerivation rec {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix
Comment (gvolpe) [reply pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix:8]: This is automatically generated running the [update.sh](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/haskell/dconf2nix/update.sh) script.

File: pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix:11]:
```suggestion
```

repo has not submodules.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124398` at 4ea29081 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>dconf2nix</li>
    <li>haskellPackages.dconf2nix</li>
  </ul>
</details>

File: pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix
Comment (SuperSandro2000) [reply pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix:8]: That script can entirely be replaced with nix-update.

Comment (gvolpe) [reply pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix:8]: Probably, but this PR only intends to update the version, not refactoring everything. That should be done in a different PR IMO.

Comment (SuperSandro2000) [reply pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix:8]: Everything is a bit of a exaggeration. I only asked to use a fitting fetcher. Also fetching commits when there is a tag available is a really bad practice. GitHub mixes commits of all forks and updating with tags through nix-update is just much easier and the current standard how to do such things.

Comment (gvolpe) [reply pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix:8]: Dude. You're asking me to fix `cabal2nix`, which has nothing to do with this PR. The `update.sh` was originally adapted from https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/purescript/spago/update.sh 

If it's so much easier you can surely put up a PR to "do it properly" later on. Now can you please stop the BS and unblock this PR that only updates the version? Thanks.

Comment (zimbatm) [reply pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix:8]: @SuperSandro2000: leaving comments on code that is not touched by the user is counter-productive. As a submitter, I should be allowed to submit a simple version bump and not have my PR blocked by unrelated issues.

For styling issues, it's best to write tools that can automate these fixes.

Comment (SuperSandro2000) [reply pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix:8]: That code is touched by this PR and this is not a styling issue but a workflow issue which causes updates to be unnecessarily hard and none automated. Also fetching commits makes reviews harder because someone needs to check if that commit actually belongs to the repo. When fetching tags this is not necessary.

Comment (zimbatm) [reply pkgs/development/tools/haskell/dconf2nix/dconf2nix.nix:8]: There is a time and place to do things.

If the PR is just a simple version bump, what the contributor cares about is getting the newer version in. It's demotivating when code reviews expand the scope of a PR and delays the merge.

A good metric is to ask yourself if the submitted code is better than the previous code.

You're also right, it would be best if `fetchFromGitHub` and the git tag were being used. For that, the best thing would be to talk to the Haskell team to see if they can add a proper heuristic to cabal2nix.

**Pull request merged**
